modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/prepare/IgniteTypeCoercion.java

```diff
@@ -762,7 +763,10 @@ private boolean validateCoerceColumn(SqlDynamicParam dynamicParam, RelDataType t
     // TODO: https://issues.apache.org/jira/browse/IGNITE-19721 - move this check to SqlValidator (if possible).
     private void validateAssignment(SqlDynamicParam node, RelDataType targetType, ContextType ctxType, IgniteSqlValidator validator) {
         RelDataType paramType = validator.resolveDynamicParameterType(node, targetType);
-        boolean compatible = TypeUtils.typeFamiliesAreCompatible(typeFactory, targetType, paramType);
+
+        boolean compatible = TypeUtils.typeFamiliesAreCompatible(typeFactory, targetType, paramType)
```

Comment:
If `TypeUtils.typeFamiliesAreCompatible ` is going to used  together with `IgniteCustomAssignmentsRules` , then it is better to extract them into a separate method that conditionally applies `IgniteCustomAssignmentsRules `.